src/include/txn_inline.h

```diff
@@ -338,15 +356,33 @@ __wt_txn_op_delete_commit_apply_timestamps(WT_SESSION_IMPL *session, WT_REF *ref
                 (*updp)->durable_ts = txn->durable_timestamp;
             }
     }
-    page_del = ref->page_del;
-    if (page_del != NULL && page_del->timestamp == WT_TS_NONE) {
-        page_del->timestamp = txn->commit_timestamp;
-        page_del->durable_timestamp = txn->durable_timestamp;
-    }
+
+    __wt_txn_op_delete_commit_apply_page_del_timestamp(session, ref);
 
     WT_REF_UNLOCK(ref, previous_state);
 }
 
+/*
+ * __wt_txn_should_assign_timestamp --
+ *     Updates without a commit time and logged objects don't have timestamps, and only the most
+ *     recently committed data matches files on disk.
+ */
+static inline bool
+__wt_txn_should_assign_timestamp(WT_SESSION_IMPL *session, WT_TXN_OP *op)
+{
+    WT_BTREE *btree;
+    WT_TXN *txn;
+
+    btree = op->btree;
+    txn = session->txn;
```

Comment:
It doesn't seem worthwhile to pull these into local variables for their single usage. The `btree` one is interesting, since the transaction operation isn't the "normal" place to retrieve a btree (that is the session handle).

Anyhow, this is a style question, but my choice here would be to change the function to be:

```F_ISSET(session->txn, WT_TXN_HAS_TS_COMMIT)```

and

```F_ISSET(op->btree, WT_BTREE_LOGGED)```

Rather than using local variables.

Reply:
Should `F_ISSET(op->btree, WT_BTREE_LOGGED)` be `F_ISSET(S2BT(session), WT_BTREE_LOGGED)` instead and remove the op argument?